src/views/token-swap/components/swap-form/swap-form.view.jsx

```diff
@@ -36,36 +59,97 @@ function SwapForm ({
   }, [accounts])
 
   React.useEffect(() => {
-    const from = accounts?.data.accounts.find(a => a.accountIndex === fromQuery)
-    const to = accounts?.data.accounts.find(a => a.accountIndex === toQuery)
-    setSelectedTokens({
-      [AmountBoxPosition.FROM]: from,
-      [AmountBoxPosition.TO]: to
-    })
+    const from = accounts.data?.accounts.find(a => a.accountIndex === fromQuery)
+    const to = accounts.data?.accounts.find(a => a.accountIndex === toQuery)
+    if (from && to) {
+      onSelectedTokensChange({
+        [AmountBoxPosition.FROM]: from,
+        [AmountBoxPosition.TO]: to
+      })
+    }
   }, [accounts])
 
-  const switchTokens = () => {
-    setSelectedTokens({
+  React.useEffect(() => {
+    if (quotes.data) {
+      const from = BigNumber.from(quotes.data[0].amountFromToken)
+      const to = BigNumber.from(quotes.data[0].amountToToken)
+      onAmountFromChange(from)
+      onAmountToChange(to)
+      if (positionUpdated === AmountBoxPosition.TO) {
+        setDefaultValues({
+          ...defaultValues,
+          [AmountBoxPosition.FROM]: { amount: from }
+        })
+      } else {
+        setDefaultValues({
+          ...defaultValues,
+          [AmountBoxPosition.TO]: { amount: to }
+        })
+      }
+    }
+  }, [quotes])
+
+  const handleAmountChange = (value, position) => {
+    clearTimeout(timer)
+    if (
+      selectedTokens.from &&
+      selectedTokens.to &&
+      positionUpdated === position
+    ) {
+      onLoadingQuotes()
+      let data = {
```

Comment:
why is this a let? maybe we can use a ternary

Reply:
Normally I prefer that flow that a ternary but if we prefer ternaries i can change it

Reply:
I agree with @AlbertoElias. We prioritize always immutability.